src/ol_infrastructure/lib/vault.py

```diff
@@ -17,134 +18,292 @@
 from ol_infrastructure.lib.pulumi_helper import StackInfo
 
 postgres_role_statements = {
-    "approle": {
-        "create": Template("CREATE ROLE ${app_name};"),
-        "revoke": Template("DROP ROLE ${app_name};"),
-    },
     "admin": {
-        "create": Template(
-            """CREATE USER "{{name}}" WITH PASSWORD '{{password}}'
-             VALID UNTIL '{{expiration}}' IN ROLE "rds_superuser"
-             INHERIT CREATEROLE CREATEDB;
-          GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "{{name}}"
-             WITH GRANT OPTION;
-          GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO "{{name}}"
-             WITH GRANT OPTION;"""
-        ),
-        "revoke": Template(
-            """REVOKE "${app_name}" FROM "{{name}}";
-          GRANT CREATE ON SCHEMA public TO ${app_name} WITH GRANT OPTION;
-          GRANT "{{name}}" TO ${app_name} WITH ADMIN OPTION;
-          SET ROLE ${app_name};
-          REASSIGN OWNED BY "{{name}}" TO "${app_name}";
-          RESET ROLE;
-          DROP OWNED BY "{{name}}";
-          REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA public FROM "{{name}}";
-          REVOKE ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public FROM "{{name}}";
-          REVOKE USAGE ON SCHEMA public FROM "{{name}}";
-          DROP USER "{{name}}";"""
-        ),
+        "create": [  # We use the prexisting rds_superuser role for admin users
+            Template(
+                """
+                CREATE USER "{{name}}" WITH PASSWORD '{{password}}'
+                VALID UNTIL '{{expiration}}' IN ROLE "rds_superuser"
+                INHERIT CREATEROLE CREATEDB;
+                """
+            ),
+            Template(
+                """
+                GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "{{name}}"
+                WITH GRANT OPTION;
+                """
+            ),
+            Template(
+                """
+                GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO "{{name}}"
+                WITH GRANT OPTION;
+                """
+            ),
+        ],
+        "revoke": [
+            # Remove the user from the pre-existing rds_superuser role
+            Template("""REVOKE "rds_superuser" FROM "{{name}}";"""),
+            # Ensure that the application role can do creations
+            Template(
+                """GRANT CREATE ON SCHEMA public TO ${app_name} WITH GRANT OPTION;"""
```

Comment:
Is this copy 🍝 from the app role? I wouldn't imagine that we want to assign admin objects to the app role.

Reply:
I'm not entirely sure where this came from. I went through the admin revoke and tried to make it make sense so it could probably use another close review with the next commit.